src/components/TwitterButton/index.js

```diff
@@ -3,40 +3,40 @@ import propTypes from 'prop-types';
 import { compose } from 'recompose';
 import { withRouter, Redirect } from 'react-router-dom';
 import { withFirebase } from '../../containers/Firebase';
-import { ReactComponent as FacebookImg } from '../../containers/assets/images/facebook.svg';
+import { ReactComponent as TwitterImage } from '../../containers/assets/images/twitter.svg';
 import * as ROUTES from '../../constants/routes';
 import './style.css';
 
 class SignInTwitter extends Component {
   state = { error: null };
 
-  handleClick = () => {
+  handleClick = async () => {
     const {
       firebase,
       history: { push },
     } = this.props;
-    firebase
-      .doSignInWithTwitter()
-      .then(socialAuthUser => {
-        firebase.user(socialAuthUser.user.uid).set(
-          {
-            email: socialAuthUser.user.email,
-            name: socialAuthUser.user.displayName,
-            goal: '',
-            userID: socialAuthUser.user.uid,
-            createdByTwitter: true,
-          },
-          { merge: true }
-        );
-        return localStorage.setItem('userId', socialAuthUser.user.uid);
-      })
-      .then(() => {
-        push(ROUTES.HOME);
-        this.setState({ error: null });
-      })
-      .catch(error => {
-        this.setState({ error });
-      });
+
+    try {
+      const socialAuthUser = await firebase.doSignInWithTwitter();
+      const userInfo = socialAuthUser.additionalUserInfo.profile;
+
+      await firebase.user(socialAuthUser.user.uid).set(
+        {
+          email: userInfo.email,
+          name: userInfo.name,
+          userID: socialAuthUser.user.uid,
+          goal: '',
+          createdByTwitter: true,
+        },
+        { merge: true }
+      );
+      localStorage.setItem('userId', socialAuthUser.user.uid);
+
+      push(ROUTES.HOME);
+      this.setState({ error: null });
```

Comment:
if we don't have errors there is no need to set the state to it 
but if you mean to empty the error message after making a new request you should put this line 
inline 14